### PR TITLE
Bar 135 local models for categorical phenotypes

### PR DESCRIPTION
**Pull request by satyarth**

### CODE REVIEW

File: src/nn/models.py [hunk 1]
```diff
@@ -361,4 +361,13 @@ def set_covariate_weights(self, weights: numpy.ndarray):
...
         weight[:, snp_count:] = cov_weights
         self.layer.weight = torch.nn.Parameter(weight)
-    
\ No newline at end of file
+
+class LassoNetClassifier(LassoNetRegressor):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        
+    def calculate_loss(self, y_hat: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
```
Comment (dkolobok) [inline src/nn/models.py:369]: maybe make a parameter for the loss with the default of binary_cross_entropy_with_logits? in case we have smth discrete but non-binary.

Comment (dkolobok) [reply src/nn/models.py:369]: and subtract 1 only if loss ==  binary_cross_entropy_with_logits
and if it's smth else, raise NotImplementedError ?

File: src/nn/models.py [hunk 2]
```diff
@@ -361,4 +361,13 @@ def set_covariate_weights(self, weights: numpy.ndarray):
...
+        y = y.unsqueeze(1).tile(dims=(1, self.hidden_size))
+        y -= 1 # BCE expects targets in the range [0, 1]
+        y_hat -= 1
```
Comment (dkolobok) [inline src/nn/models.py:372]: do we need to subtract 1 here as well? I thought it should be two values for probabilities for 2 output neurons, no?

File: src/nn/models.py [hunk 3]
```diff
@@ -361,4 +361,13 @@ def set_covariate_weights(self, weights: numpy.ndarray):
...
-    
\ No newline at end of file
+
+class LassoNetClassifier(LassoNetRegressor):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        
+    def calculate_loss(self, y_hat: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        y = y.unsqueeze(1).tile(dims=(1, self.hidden_size))
+        y -= 1 # BCE expects targets in the range [0, 1]
```
Comment (dkolobok) [inline src/nn/models.py:371]: maybe add an assert that observed values are coded as 1 and 2?

File: src/preprocess/phenotypes_covariates.py [hunk 4]
```diff
@@ -2,18 +2,31 @@
...
-    loader = UKBDataLoader(ukb_dataset_path, 'split', str(phenotype_code), list(covariates.keys()))
-    pheno_cov = pandas.concat((loader.load_train(), loader.load_val(), loader.load_test()))
-    pheno_cov.rename({str(phenotype_code) : phenotype_name}, axis='columns', inplace=True)
+    if phenotype_type == 'real':
+        loader = UKBDataLoader(ukb_dataset_path, 'split', str(phenotype_code), list(covariates.keys()))
+        pheno_cov = pandas.concat((loader.load_train(), loader.load_val(), loader.load_test()))
+        pheno_cov.rename({str(phenotype_code) : phenotype_name}, axis='columns', inplace=True)
+    
+    elif phenotype_type == 'binary':
+        sd_field_code = '20002'
```
Comment (dkolobok) [inline src/preprocess/phenotypes_covariates.py:21]: why is 20002 hardcoded?
